src/routers/controllers/manageUsersController.ts

```diff
@@ -80,69 +80,66 @@ export const getViewData = async (req: Request): Promise<AnyRecord> => {
+            } else {
                 viewData.manageUsersTabId = constants.ACCOUNT_OWNERS_TAB_ID;
             }
-        } else {
-            errorMessage = constants.ERRORS_ENTER_AN_EMAIL_ADDRESS_IN_THE_CORRECT_FORMAT;
-            viewData.errors = {
-                search: {
-                    text: errorMessage
-                }
-            };
-            pageNumbers.ownerPage = 1;
-            pageNumbers.adminPage = 1;
-            pageNumbers.standardPage = 1;
+        } catch (error) {
+            logger.error(`ACSP membership for email ${search} not found.`);
+            viewData.manageUsersTabId = constants.ACCOUNT_OWNERS_TAB_ID;
         }
-        viewData.search = search;
+    } else if (search) {
+        errorMessage = constants.ERRORS_ENTER_AN_EMAIL_ADDRESS_IN_THE_CORRECT_FORMAT;
+        viewData.errors = {
+            search: {
+                text: errorMessage
+            }
+        };
+        pageNumbers.ownerPage = 1;
+        pageNumbers.adminPage = 1;
+        pageNumbers.standardPage = 1;
     }
 
-    const ownerMemberRawViewData = await getMemberRawViewData(req, acspNumber, pageNumbers, UserRole.OWNER, constants.ACCOUNT_OWNERS_TAB_ID, translations);
-    ownerMembers = ownerMemberRawViewData.memberships;
-    viewData.accoutOwnerPadinationData = ownerMemberRawViewData.pagination;
-
-    const adminMemberRawViewData = await getMemberRawViewData(req, acspNumber, pageNumbers, UserRole.ADMIN, constants.ADMINISTRATORS_TAB_ID, translations);
-    adminMembers = adminMemberRawViewData.memberships;
-    viewData.adminPadinationData = adminMemberRawViewData.pagination;
-
-    const standardMemberRawViewData = await getMemberRawViewData(req, acspNumber, pageNumbers, UserRole.STANDARD, constants.STANDARD_USERS_TAB_ID, translations);
-    standardMembers = standardMemberRawViewData.memberships;
-    viewData.standardUserPadinationData = standardMemberRawViewData.pagination;
+    if (search) {
+        viewData.search = search;
+    } else {
+        // Only fetch all users if there's no search
+        const ownerMemberRawViewData = await getMemberRawViewData(req, acspNumber, pageNumbers, UserRole.OWNER, constants.ACCOUNT_OWNERS_TAB_ID, translations);
+        ownerMembers = ownerMemberRawViewData.memberships;
+        viewData.accoutOwnerPadinationData = ownerMemberRawViewData.pagination;
+
+        const adminMemberRawViewData = await getMemberRawViewData(req, acspNumber, pageNumbers, UserRole.ADMIN, constants.ADMINISTRATORS_TAB_ID, translations);
+        adminMembers = adminMemberRawViewData.memberships;
+        viewData.adminPadinationData = adminMemberRawViewData.pagination;
+
+        const standardMemberRawViewData = await getMemberRawViewData(req, acspNumber, pageNumbers, UserRole.STANDARD, constants.STANDARD_USERS_TAB_ID, translations);
+        standardMembers = standardMemberRawViewData.memberships;
+        viewData.standardUserPadinationData = standardMemberRawViewData.pagination;
+    }
```

Comment:
Just a thought, is there a potential performance improvement if we use `await Promise.all([...])` rather than sequentially calling multiple awaits?

Reply:
Hey - good point.  I did this and it actually made it noticeabley quicker - https://github.com/companieshouse/acsp-manage-users-web/pull/102/commits/5cfe8b334d71196a90cc8df0364973e75214ffd2

Could you re-review/approve when you have time please?

Reply:
Loooks good